crates/util/cometindex/src/indexer.rs

```diff
@@ -21,8 +22,13 @@ async fn catchup(
     tracing::info!(?src_height, ?index_heights, "catchup status");
     let lowest_index_height = index_heights.values().copied().min().unwrap_or_default();
     if lowest_index_height >= src_height {
-        tracing::info!(why = "already caught up", "catchup completed");
-        return Ok(());
+        if exit_on_catchup {
+            tracing::info!("catchup completed, exiting as requested");
+            std::process::exit(0);
```

Comment:
It's a bit of a hack to `std::process::exit()` here. Perhaps we should instead return a custom Result::CatchUp that triggers an exit in the main loop? Didn't seem worthwhile for a quick feature add, but open to suggestions.

Reply:
I wrote up a quick commit to avoid std::process::exit: https://github.com/penumbra-zone/penumbra/commit/da002eebf061f0d9f7f392ef3b13a08f85c802bd ; I think I'd like to see that included in this PR (or something equivalent). `Result<bool>`  seems fine enough to capture the possibilities there, and avoids the woes of direct exits, in case we want to run this code in a different context.